sensors/gps/ubx.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Driver for GPS module ubx
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau, Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <math.h>
+#include <string.h>
+#include <termios.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+#include "common.h"
+#include "../sensors.h"
+#include "nmea.h"
+#include "ubx.h"
+
+#define ubx_STR "ubx:"
+
+/* Specified by the module's documentation */
+#define UPDATE_RATE_MS 100
+
+/* ubx has: pos_CEP = 3m, vel_CEP = 0.1m/s. We use RMS values which are 0.849 * CEP (https://en.wikipedia.org/wiki/Circular_error_probable) */
+#define UBX_POS_ACCURACY 2.547f
+#define UBX_VEL_ACCURACY 0.0849f
+
+#define REC_BUF_SZ 1024
+#define INBOX_SIZE 3
+
+
+typedef struct {
+	sensor_event_t evtGps;
+	int filedes;
+	gps_receiver_t receiver;
+	char stack[1024] __attribute__((aligned(8)));
```

Comment:
This stack might prove to be too small, `printf` alone uses ~400 bytes of stack

Reply:
Increased to 2048